### ISSUE TITLE
Spurious send tries in backend

### ISSUE THREAD

**Issue by julianoes** [Dronecode/DronecodeSDK#609]
When the backend is started, something tries to send messages even though we have not discovered a vehicle.

The output is this:
```
[03:39:09|Info ] DronecodeSDK version: 0.9.0 (dronecode_sdk_impl.cpp:24)
[03:39:09|Debug] New: System ID: 0 Comp ID: 0 (dronecode_sdk_impl.cpp:284)
[03:39:09|Info ] Server set to listen on 0.0.0.0:50051 (grpc_server.cpp:40)
[03:39:09|Info ] Server started (grpc_server.cpp:24)
[03:39:09|Info ] Waiting to discover system... (connection_initiator.h:57)
[03:39:10|Error] No known remotes (udp_connection.cpp:130)
[03:39:10|Error] send fail (dronecode_sdk_impl.cpp:90)
[03:39:11|Error] No known remotes (udp_connection.cpp:130)
[03:39:11|Error] send fail (dronecode_sdk_impl.cpp:90)
[03:39:12|Error] No known remotes (udp_connection.cpp:130)
[03:39:12|Error] send fail (dronecode_sdk_impl.cpp:90)
[03:39:13|Error] No known remotes (udp_connection.cpp:130)
[03:39:13|Error] send fail (dronecode_sdk_impl.cpp:90)
[03:39:14|Error] No known remotes (udp_connection.cpp:130)
[03:39:14|Error] send fail (dronecode_sdk_impl.cpp:90)
[03:39:15|Error] No known remotes (udp_connection.cpp:130)
[03:39:15|Error] send fail (dronecode_sdk_impl.cpp:90)
[03:39:16|Error] No known remotes (udp_connection.cpp:130)
[03:39:16|Error] send fail (dronecode_sdk_impl.cpp:90)
[03:39:17|Error] No known remotes (udp_connection.cpp:130)
[03:39:17|Error] send fail (dronecode_sdk_impl.cpp:90)
[03:39:18|Error] No known remotes (udp_connection.cpp:130)
[03:39:18|Error] send fail (dronecode_sdk_impl.cpp:90)
[03:39:19|Error] No known remotes (udp_connection.cpp:130)
[03:39:19|Error] send fail (dronecode_sdk_impl.cpp:90)
[03:39:20|Error] No known remotes (udp_connection.cpp:130)
[03:39:20|Error] send fail (dronecode_sdk_impl.cpp:90)
[03:39:21|Error] No known remotes (udp_connection.cpp:130)
[03:39:21|Error] send fail (dronecode_sdk_impl.cpp:90)
```

Steps to reproduce:
1. Don't run PX4 SITL, make sure nothing would connect.
2. `make BUILD_BACKEND=1 && build/default/backend/src/backend_bin`

**Comment by jarednm** [comment]
Any update on this?  I have the same issue.

**Comment by JonasVautherin** [comment]
As it is a "beginner" issue, we're keeping it open, hoping that somebody will feel like contributing :-)

**Comment by snail914** [comment]
Bug found^^ My laptop is too slow to build the project, will send a pr tomorrow

**Comment by hamishwillee** [comment]
You know that MAVLink components should publish heartbeats even if they haven't discovered other systems yet, right?

**Comment by julianoes** [comment]
> You know that MAVLink components should publish heartbeats even if they haven't discovered other systems yet, right?

From my understanding that's not true. Systems such as drones should be sending heartbeats and ground stations are supposed to listen for heartbeats. Once they receive something, they can (don't have to) respond using heartbeats.

**Comment by hamishwillee** [comment]
>> You know that MAVLink components should publish heartbeats even if they haven't discovered other systems yet, right?

> From my understanding that's not true. Systems such as drones should be sending heartbeats and ground stations are supposed to listen for heartbeats. Once they receive something, they can (don't have to) respond using heartbeats.

Not what the [spec says](https://mavlink.io/en/services/heartbeat.html#connectiondisconnection). Which does not mean you are wrong, but that we need this confirmed. 
I will confirm in dev call.

**Comment by julianoes** [comment]
As far as I know, QGC does not do it either, at least on UDP because it would not even know where to send heartbeats. The port was not defined until recently when it was written in the port diagram. The other example is MAVROS which does send stuff by assuming the SITL port which just always happened to be the same (14557 I think).

**Comment by hamishwillee** [comment]
As discussed in devcall. You have a good point, with the result that you now have action to work out best way to handle connection, vehicle discovery and system id allocation mechanisms. That'll learn ya!

**Comment by julianoes** [comment]
Fixed by #674.
